.github/workflows/ecs.yml

```diff
@@ -0,0 +1,81 @@
+name: ecs
+
+on:
+  workflow_call:
+    inputs:
+      composer-command:
+        description: Composer command to run.
+        default: composer update --prefer-dist --no-interaction --no-progress --optimize-autoloader --ansi
+        required: false
+        type: string
+      composer-version:
+        description: Composer version to use.
+        default:
+        required: false
+        type: string
+      coverage:
+        description: Generate code coverage report.
+        default: none
+        required: false
+        type: string
+      extensions:
+        description: List of extensions to PHP.
+        default:
+        required: false
+        type: string
+      ini-values:
+        description: Initial values for PHP configuration.
+        default: date.timezone='UTC'
+        required: false
+        type: string
+      os:
+        description: Os to test, separated by comma.
+        required: true
+        type: string
+      php:
+        description: PHP versions to test, separated by comma.
+        required: true
+        type: string
+      tools:
+        description: Tools to test, separated by comma.
+        default: composer:v2
+        required: false
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  ecs:
+    name: PHP ${{ matrix.php }}-${{ matrix.os }}
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ${{ fromJson(inputs.os) }}
+        php: ${{ fromJson(inputs.php) }}
+
+    steps:
+      - name: Checkout.
+        uses: actions/checkout@v4
+
+      - name: Install PHP.
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: ${{ inputs.coverage }}
+          extensions: ${{ inputs.extensions}}
+          ini-values: ${{ inputs.ini-values }}
+          php-version: ${{ matrix.php }}
+          tools: ${{ inputs.tools }}
+
+      - name: Update composer.
+        run: composer self-update ${{ inputs.composer-version }}
+
+      - name: Install dependencies with composer.
+        run: ${{ inputs.composer-command }}
+
+      - name: Run easy-coding-standard.
+        run: vendor/bin/ecs check src tests --ansi
```

Comment:
Need autofix.

Reply:
no.

Reply:
Yes) Make it optional via action configuration.

Reply:
But we will have the same problem with the rector for commits, and i don't want to drag that here.

Reply:
Add this optional and disabled by default

Reply:
I prefer to do it in a separate PR, and try to solve the rector's problem once and for all.

Reply:
We need auto-fix. That's why StyleCI fits us well. I don't want contributors to spend time on fixing code style issues at all.